File: online-judge/haskell/ex001.hs
{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}

import Data.Bits (xor)

-- https://oj.moonbitlang.com/problems/202412-001-find-the-unique-number

{-
>>> solution([1, 1, 2, 2, 3, 3, 4, 5, 5])
4

>>> solution([0, 1, 0, 1, 2])
2

>>> solution([7, 3, 3, 7, 10])
10
 -}
solution :: [Int] -> Int
solution = foldr (flip xor) 0

File: online-judge/rust/tests/ex001.rs
/*

https://oj.moonbitlang.com/problems/202412-001-find-the-unique-number

*/

fn solution(nums: Vec<i32>) -> i32 {
    nums.iter().fold(0, |acc, x| acc ^ x)
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_solution() {
        assert_eq!(4, solution(vec![1, 1, 2, 2, 3, 3, 4, 5, 5]));
        assert_eq!(2, solution(vec![0, 1, 0, 1, 2]));
        assert_eq!(10, solution(vec![7, 3, 3, 7, 10]));
    }
}
